File: FrontEnd/barbearia/src/models/Person.ts
export class Person {
  public id?: number | undefined
  public name: string
  public email: string
  public senha: string
  public permissao: string
  public cpf: string
  public whatsApp: number

  constructor(
    id = undefined,
    name,
    email,
    senha,
    permissao,
    cpf,
    whatsApp,
  ) {
    this.id = id
    this.name = name
    this.email = email
    this.senha = senha
    this.permissao = permissao
    this.cpf = cpf
    this.whatsApp = whatsApp
  }
}

File: API/Model/Person.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using API.Model.Base;

namespace API.Model
{
    [Table("Persons")]
    public class Person : BaseEntity
    {
        [Column("name")]
        [Required]
        [StringLength(150)]
        public string Name { get; set; }

        [Column("senha")]
        [Required]
        [Range(1, 20)]
        public string Senha { get; set; }       

        [Column("email")]
        [StringLength(300)]
        public string? Email { get; set; }
        
        public int? PermissaoId {get; set; }
        [ForeignKey("PermissaoId")]
        public virtual Permission IdPermission { get; set; }

        [Column("cpf")]
        [StringLength(20)]
        public string? Cpf { get; set; }

        [Column("whats-app")]
        [Range(1, 20)]
        public decimal? WhatsApp { get; set; }

        [Column("image_url")]
        [StringLength(300)]
        public string? ImageUrl { get; set; }

    }
}
